packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -204,34 +252,24 @@ const SettingsRoutes = ({
   integration,
 }: { isSignedIn: boolean; integration: Integration } & RouteComponentProps) => {
   const location = useLocation();
-  // TODO: Remove this + config.sendFxAStatusOnSettings check once we confirm this works
-  const config = useConfig();
+  const isSync = integration != null ? integration.isSync() : false;
 
-  // Request and update account data/state to match the browser state because:
-  // 1) If a user logs in through the browser Sync pairing flow and then accesses
-  // Settings through the browser, they'd have to login again (isSignedIn === false).
-  // 2) If a user disconnects their account through Sync and then logs into another account
-  // using the pairing flow, we hold onto the old session token (isSignedIn === true).
-  const [shouldCheckFxaStatus, setShouldCheckFxaStatus] = useState(
-    // See note above WebIntegration's `isSync` check
-    isWebIntegration(integration) &&
-      integration.isSync() &&
-      config.sendFxAStatusOnSettings
-  );
-  useEffect(() => {
-    (async () => {
-      if (shouldCheckFxaStatus) {
-        await firefox.requestSignedInUser(integration.data.context);
-        setShouldCheckFxaStatus(false);
-      }
-    })();
-  });
+  // If the user is not signed in, they cannot access settings! Direct them accordingly
+  if (!isSignedIn) {
+    const params = new URLSearchParams(window.location.search);
+
+    if (isSync) {
+      // For sync this means we somehow dropped the sign out message, which is
+      // a known issue in android. In this case, our best option is to ask the
+      // user to manually signout from sync.
+      hardNavigate(`/signout_sync?${params.toString()}`);
```

Comment:
If we do want to leave this as a new page, I think we'd just want to regular navigate.

Reply:
Just curious why does it matter, since there is no reach router state? FWIW, the other redirect, which already existed was a hard navigate, so I just kept it consistent with that. Happy to change it though.

Reply:
I just tired to changing this to navigate, and then I sometimes encounter a hanging spinner. Not sure why yet, but I've pushed the change for hardNavigate back up, so as not to block any manual testing.

Reply:
`hardNavigate` should force a full page reload (it just does a `window.location.href = whatever`) and is usually used for navigating away from React, and using regular navigate keeps the user in the SPA.

Ah, I think I see what's going on. You have `<SignoutSync path="/signout_sync" />` under Settings routes, so I think you're making it accessible at `/settings/signout_sync` instead of `signout_sync`.

Reply:
I've switched over to the component approach.